core/src/test/java/com/gentics/mesh/core/user/JWTConfigurationTest.java

```diff
@@ -0,0 +1,88 @@
+package com.gentics.mesh.core.user;
+
+import static com.gentics.mesh.test.TestSize.PROJECT;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.assertNull;
+
+import java.io.IOException;
+import java.util.Arrays;
+import java.util.Base64;
+import java.util.Collection;
+import java.util.function.Consumer;
+
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.Parameterized;
+
+import com.gentics.madl.tx.Tx;
+import com.gentics.mesh.core.data.User;
+import com.gentics.mesh.core.rest.common.GenericMessageResponse;
+import com.gentics.mesh.etc.config.MeshOptions;
+import com.gentics.mesh.rest.client.MeshRestClient;
+import com.gentics.mesh.test.context.AbstractMeshTest;
+import com.gentics.mesh.test.context.MeshTestContext;
+import com.gentics.mesh.test.context.MeshTestHelper;
+import com.gentics.mesh.test.context.MeshTestSetting;
+
+import io.reactivex.Single;
+import io.vertx.core.json.JsonArray;
+import io.vertx.core.json.JsonObject;
+
+@RunWith(Parameterized.class)
+@MeshTestSetting(testSize = PROJECT, startServer = true)
+public class JWTConfigurationTest extends AbstractMeshTest {
+
+	@Parameterized.Parameter(0)
+	public Consumer<MeshOptions> optionChanger;
+
+	@Parameterized.Parameters(name = "{index}: authOptions={0}")
+	public static Collection<Object[]> paramData() {
+		return Arrays.asList(new Object[][] {
+			{ MeshTestHelper.noopConsumer() },
+			{(Consumer<MeshOptions>) meshOptions -> {
+				meshOptions.getAuthenticationOptions()
+					.setIssuer("HELLO WORLD")
+					.setAudience(Arrays.asList("www.example.com", "auth.example.com"));
+			}}
+		});
+	}
+
+	@Override
+	public MeshTestContext getTestContext() {
+		return testContext.setOptionChanger(optionChanger);
+	}
+
+	@Test

```

Comment:
Need also an unhappy case (bad audience / bad issuer etc).

Reply:
Need also a leeway configuration test. Since the leeway setting is set explicitly on the Mesh start, it is not applied with OptionsChanger, and requires the separate test file.